platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DynamicCallProblemReporting.kt

```diff
@@ -84,9 +88,10 @@ class DefaultDynamicCallProblemReporting : DynamicCallProblemReporting {
 
     private
     fun logMissingScope() {
-        logger.debug {
-            val stackTrace = IllegalStateException("Expected unreportedProblemInCurrentCall to be called after enterDynamicCall").stackTraceToString()
-            "Warning: " + stackTrace.lines().take(15).joinToString("\n")
+        if (logger.isDebugEnabled) {
+            getUniqueLocationWithStacktrace("Expected unreportedProblemInCurrentCall to be called after enterDynamicCall")?.let {
+                logger.debug("Warning: $it")
+            }
```

Comment:
This is the main change in terms of behavior. This will print the full exception the first time it is observed, and not repeat it afterwards. The repeated exceptions do not provide additional information in this case.